lib/kafka_ex/gen_consumer.ex

```diff
@@ -262,9 +262,20 @@ defmodule KafkaEx.GenConsumer do
         {:ok, nil}
       end
 
-      def handle_call(_call, _from, _consumer_state) do
-        # the user must implement this if they expect to recieve calls
-        :handle_call_not_implemented
+      def handle_call(msg, _from, _consumer_state) do
+        # taken from the GenServer handle_call implementation
+        case Process.info(self(), :registered_name) do
+          {_, []}   -> self()
+          {_, name} -> name
+        end
+
+        # We do this to trick Dialyzer to not complain about non-local returns.
+        case :erlang.phash2(1, 1) do
+          0 ->
+            raise "attempted to call GenServer #{inspect proc} but no " <>
+              "handle_call/3 clause was provided"
+          1 -> {:reply, {:bad_call, msg}, consumer_state}
```

Comment:
consumer_state isn't a valid identifier here, wouldn't this cause a compile warning?

Reply:
:rage1:  Added a test, pushed fixes, waiting for travis.

Reply:
do we need to actually return something here, it seems this will never match at least from me running in erl, is `_ -> _ # with some comment` not adequate?

Reply:
It should never return this in practice, but I think it's necessary for it to return a value within the dialyzer spec so that dialyzer succeeds.  I assume they did this on purpose in the GenServer implementation.

Reply:
Ah I see, how about `_ -> {:reply, {:bad_call, msg}, consumer_state}` so the int `1` is less confusing, it's annoying to have write dumb code to just make dialyzer happy